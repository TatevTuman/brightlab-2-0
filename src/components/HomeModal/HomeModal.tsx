import React from 'react'
import { Modal } from '@components'
import { useModal } from '@hooks'
import styles from './HomeModal.module.scss'

export interface HomeModalProps {}

const HomeModal: React.FC<HomeModalProps> = props => {
  const modalName = 'HomeModal'
  const { modal, closeModal } = useModal(modalName)
  const number = modal?.state?.number

  return (
    <Modal modalName={modalName}>
      <div className={styles.homeModal}>
        <Modal.Header>
          <h3>Home modal</h3>
        </Modal.Header>
        <div>Random number from modal state: {number}</div>
        <Modal.Footer>
          <div onClick={closeModal}>Close</div>
        </Modal.Footer>
      </div>
    </Modal>
  )
}

export default HomeModal
