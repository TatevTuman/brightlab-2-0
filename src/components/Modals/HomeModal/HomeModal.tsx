import React, { memo } from 'react'
import { Modal } from '@components'
import { useModal } from '@hooks'
import styles from './HomeModal.module.scss'

export interface HomeModalProps {}

const HomeModal: React.FC<HomeModalProps> = props => {
  const { modal, closeModal } = useModal('HomeModal')
  const number = modal?.state?.number

  return (
    <Modal id={'HomeModal'} opened={!!modal} onClose={closeModal}>
      <div className={styles.homeModal}>
        <Modal.Header>
          <h3>Home modal</h3>
        </Modal.Header>
        <div>Random number from modal state: {number}</div>
        <Modal.Footer>
          <div onClick={closeModal}>Close</div>
        </Modal.Footer>
      </div>
    </Modal>
  )
}

export default memo(HomeModal)
