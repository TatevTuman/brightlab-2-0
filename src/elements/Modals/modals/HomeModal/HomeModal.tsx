import React, { memo } from 'react'
import { Modal } from '@modules'
import { useModal } from '@hooks'
import './HomeModal.scss'

export interface HomeModalProps {}

const HomeModal: React.FC<HomeModalProps> = props => {
  const { modal, closeModal } = useModal('HomeModal')
  const number = modal?.state?.number

  return (
    <Modal id={'HomeModal'} opened={!!modal} onClose={closeModal} width={400}>
      <div className={'home-modal'}>
        <Modal.Header>
          <h3>Home modal</h3>
        </Modal.Header>
        <div>Random number from modal state: {number}</div>
        <Modal.Footer>
          <div onClick={closeModal}>Close</div>
        </Modal.Footer>
      </div>
    </Modal>
  )
}

export default memo(HomeModal)
