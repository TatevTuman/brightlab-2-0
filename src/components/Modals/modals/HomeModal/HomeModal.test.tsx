import React, { useEffect } from 'react'
import { render, cleanup, waitFor } from '@testing-library/react'
import { useModal } from '@hooks'
import HomeModal from './HomeModal'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

const TestHomeModal = () => {
  const modal = useModal('HomeModal')

  useEffect(() => {
    modal.openModal()
  }, [])

  return (
    <>
      <div id={'modals'} />
      <HomeModal {...modal} />
    </>
  )
}

describe('HomeModal', () => {
  it('renders correctly', async () => {
    const { container } = render(<TestHomeModal />)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
  })
})
