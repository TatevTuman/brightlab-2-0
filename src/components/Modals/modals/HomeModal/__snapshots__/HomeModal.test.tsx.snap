// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeModal renders correctly 1`] = `
<div>
  <div
    id="modals"
  />
  <div
    class="modal"
    id="HomeModal"
    role="complementary"
  >
    <div
      class="modal-mask"
    />
    <div
      class="modal-content"
    >
      <div
        class="homeModal"
      >
        <header
          class="modal-header"
        >
          <h3>
            Home modal
          </h3>
        </header>
        <div>
          Random number from modal state: 
        </div>
        <footer
          class="modal-footer"
        >
          <div>
            Close
          </div>
        </footer>
      </div>
    </div>
  </div>
</div>
`;
